### PR TITLE
Fix issue with user change on a Device (#296)

### PR DESCRIPTION
**Pull request by PierreF**
This is a fix for #296 . It's the [solution](https://github.com/jleclanche/django-push-notifications/issues/296#issuecomment-225462380) proposed by @chozabu with the following modification:

* Don't do repeat the create method, use self.queryset.model to access GCMDevice or APNSDevice
* Add the setting in PUSH_NOTIFICATIONS_SETTINGS
* Don't crash if "registration_id" is not in request.data :)

### CODE REVIEW

File: push_notifications/api/rest_framework.py [hunk 1]
```diff
@@ -125,6 +128,22 @@ def has_object_permission(self, request, view, obj):
...
+			).first()
+			if instance:
+				serializer = self.get_serializer(instance, data=request.data)
+		if not serializer:
+			serializer = self.get_serializer(data=request.data)
+
+		serializer.is_valid(raise_exception=True)
+		self.perform_create(serializer)
+		headers = self.get_success_headers(serializer.data)
+		return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
```
Comment (jamaalscarlett) [inline push_notifications/api/rest_framework.py]: Looks good, but in the case of an update the http status code should be 200 or 204

**Comment by PierreF** [comment]
Updated the PR to return a status code 200 in case of update (as do the default update mixing of rest framwork).

**Comment by jleclanche** [comment]
Not a fan of the setting's name. Other than than, lgtm but I'll leave it up to @jamaalscarlett to decide.

**Comment by PierreF** [comment]
UPDATE_ON_DUPLICATE_REG_ID ?
UPDATE_DEVICE_SAME_REG_ID  ?

If someone has a better name, I'll take it :)

**Comment by jamaalscarlett** [comment]
I'm fine with UPDATE_ON_DUPLICATE_REG_ID, let's go with that one.

**Comment by jleclanche** [comment]
Thanks!

**Pull request merged**